examples/statistics/customized_violin_demo.py

```diff
@@ -0,0 +1,77 @@
+# Customizing violin plots
+#
+#
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+# functions to calculate percentiles and adjacent values
+def percentile(vals, p):
+    N = len(vals)
+    n = p*(N+1)
+    k = int(n)
+    d = n-k
+    if k <= 0:
+        return vals[0]
+    if k >= N:
+        return vals[N-1]
+    return vals[k-1] + d*(vals[k] - vals[k-1])
+
+
+def adjacent_values(vals):
+    q1 = percentile(vals, 0.25)
+    q3 = percentile(vals, 0.75)
+    uav = q3 + (q3-q1)*1.5
+    if uav > vals[-1]:
+        uav = vals[-1]
+    if uav < q3:
+        uav = q3
+    lav = q1 - (q3-q1)*1.5
+    if lav < vals[0]:
+        lav = vals[0]
+    if lav > q1:
+        lav = q1
+    return [lav, uav]
+
+# create test data
+dat = [np.random.normal(0, std, 100) for std in range(6, 10)]
+lab = ['a', 'b', 'c', 'd']    # labels
+med = []    # medians
+iqr = []    # inter-quantile ranges
+avs = []    # upper and lower adjacent values
+for arr in dat:
+    sarr = sorted(arr)
+    med.append(percentile(sarr, 0.5))
+    iqr.append([percentile(sarr, 0.25), percentile(sarr, 0.75)])
+    avs.append(adjacent_values(sarr))
+
+# plot the violins
+fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(7, 5))
+parts = ax.violinplot(dat, showmeans=False, showmedians=False,
+                      showextrema=False)
+
+# plot medians and averages
+for i in range(len(med)):
```

Comment:
You can do the lines with a single call to `ax.vlines` and all of the markes with a single call to `ax.plot`.  On the other hand, if it is the idea to be able to make a single violin invisible  at a time this way might be better.

Can you collect the returned artists into a data structure of some sort (probably a dictionary per violin?)?